shared/src/main/scala/coop/rchain/shared/RChainEvent.scala

```diff
@@ -0,0 +1,61 @@
+package coop.rchain.shared
+
+import cats.Applicative
+
+sealed trait RChainEvent {}
+
+final case class BlockCreated(
+    blockHash: String,
+    parentHashes: List[String],
+    justificationHashes: List[(String, String)],
+    deployIds: List[String],
+    creator: String,
+    seqNum: Int
+) extends RChainEvent
+
+final case class BlockAdded(
+    blockHash: String,
+    parentHashes: List[String],
+    justificationHashes: List[(String, String)],
+    deployIds: List[String],
+    creator: String,
+    seqNum: Int
+) extends RChainEvent
+
+final case class BlockFinalised(blockHash: String) extends RChainEvent
+
+object RChainEvent {
+  def created(
+      bs: String,
+      parents: List[String],
+      justifications: List[(String, String)],
+      deployIds: List[String],
+      creator: String,
+      seqNum: Int
+  ): RChainEvent =
+    BlockCreated(bs, parents, justifications, deployIds, creator, seqNum)
+
+  def added(
+      bs: String,
+      parents: List[String],
+      justifications: List[(String, String)],
+      deployIds: List[String],
+      creator: String,
+      seqNum: Int
+  ): RChainEvent =
+    BlockAdded(bs, parents, justifications, deployIds, creator, seqNum)
+
+  def finalised(bs: => String): RChainEvent = BlockFinalised(bs)
+}
```

Comment:
I would prefer the full event names (in camel case) as the constructor names:
- `blockCreated`
- `blockAdded`